### PR TITLE
Fix crash when handling non-food liquids

### LINKED ISSUE #28704: Siphoning gas to/from vehicle crashes the game, sometimes
**Describe the bug**  
Siphoning diesel from a cargo truck into a plastic jerrycan causes a crash. Siphonong from same truck into a gallon jug doesn't crash

CRASH LOG FILE: config/crash.log
VERSION: 0.D-498-g5b6a055
TYPE: Signal
MESSAGE: SIGABRT: Abnormal termination
STACK TRACE:
    @0x554E45[cataclysm-tiles.exe+0x154E45]
    @0x5559F2[cataclysm-tiles.exe+0x1559F2]
    raise+0x200@0x7FFB6479A9E0[msvcrt.dll+0x2A9E0]
    abort+0x1B@0x7FFB6479EDEB[msvcrt.dll+0x2EDEB]
    IMG_LoadWEBP_RW+0xFDCC1@0x100A2E1[cataclysm-tiles.exe+0xC0A2E1]
    IMG_LoadWEBP_RW+0xFDDA7@0x100A3C7[cataclysm-tiles.exe+0xC0A3C7]
    @0x71F499[cataclysm-tiles.exe+0x31F499]
    @0x73B6D5[cataclysm-tiles.exe+0x33B6D5]
    @0x4E7078[cataclysm-tiles.exe+0xE7078]
    @0x427CFA[cataclysm-tiles.exe+0x27CFA]
    @0x432AEC[cataclysm-tiles.exe+0x32AEC]
    @0xB7FF3B[cataclysm-tiles.exe+0x77FF3B]
    @0x646EA4[cataclysm-tiles.exe+0x246EA4]
    @0x685D7C[cataclysm-tiles.exe+0x285D7C]
    IMG_LoadWEBP_RW+0x4D2619@0x13DEC39[cataclysm-tiles.exe+0xFDEC39]
    @0x4013ED[cataclysm-tiles.exe+0x13ED]
    @0x4014FB[cataclysm-tiles.exe+0x14FB]
    BaseThreadInitThunk+0x14@0x7FFB663E3DC4[KERNEL32.DLL+0x13DC4]
    RtlUserThreadStart+0x21@0x7FFB66943691[ntdll.dll+0x73691]

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary

 ```SUMMARY: Bugfixes "Fix crash when handling non-food liquids"```

Fixes #28704

#### Purpose of change
When you tried to put non food liquid into a container that was not empty the game tried to average their temperatures.

But since they were not food and had no temperature data at all this resulted the game to crash.

#### Describe the solution
Check if the container has food in it before trying to handle its contents as food.

#### Describe alternatives you've considered
-

#### Additional context
-

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -6297,7 +6297,7 @@ void item::fill_with( item &liquid, long amount )
             return;
         }
         ammo_set( liquid.typeId(), ammo_remaining() + amount );
-    } else if( !is_container_empty() ) {
+    } else if( is_food_container() ) {
```
Comment (Night-Pryanik) [inline src/item.cpp]: Removing the check for non-empty container will try to calculate the temperature even if there is no drinkable liquid.

Comment (Hirmuolio) [reply src/item.cpp]: `is_food_container()` returns false if the container is empty.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/crash-while-siphoning-fuel/19234/2

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/crashes-bugs-etc/19233/2

**Pull request merged**